src/App.js

```diff
@@ -14,11 +14,83 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGood: ['Jam'],
+  }
+
+  title = (param) => {
+    switch (param.length) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return `${param[0]} is selected`;
+      default:
+        return `${param.slice(0, -1).join(', ')}`
+        + ` and ${param[param.length - 1]} is selected `;
+    }
+  }
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          {this.title(selectedGood)}
+          <button
+            type="button"
+            className={selectedGood.length === 0
+              ? 'hide'
+              : ''}
+            onClick={() => {
+              this.setState({ selectedGood: [] });
+            }}
+          >
+            X
+          </button>
+        </h1>
+
+        <ul>
+          {goodsFromServer.map(good => (
+            <>
+              <li
+                key={good}
+                className={selectedGood.includes(good)
```

Comment:
you can create some variable (`isIncluded` for example) inside callback in `map` method before you return `<li>`, and use this variable in all places where you need it. with this approach, you can use the method includes only once